src/cases/repositories/case.repository.js

```diff
@@ -10,7 +10,28 @@ export const findById = async (caseId) => {
   return payload;
 };
 
-export const updateStage = async (caseId) => {
-  const { payload } = await wreck.post(`/cases/${caseId}/stage`);
-  return payload;
+export const updateTaskStatus = async ({
+  caseId,
+  stageId,
+  taskGroupId,
+  taskId,
+  isComplete,
+}) => {
+  return await wreck.patch(
+    `/cases/${caseId}/stages/${stageId}/task-groups/${taskGroupId}/tasks/${taskId}/status`,
+    {
+      payload: { status: isComplete ? "complete" : "pending" },
+    },
+  );
+};
+
+export const completeStage = async (caseId) => {
+  try {
+    await wreck.post(`/cases/${caseId}/stage`);
+  } catch (e) {
+    if (e.data?.payload) {
+      return { error: e.data.payload };
+    }
+    return { error: "Update failed" };
```

Comment:
As we don't yet have validation and UX for displaying the error, let's let it fail with an error page. I.e. not catch errors from the POST